src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.exeption.DataProcessingExeption;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl extends GenericDaoImpl<ShoppingCart> implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart entity) {
+        return super.add(entity);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM ShoppingCart sc "
+                    + "LEFT JOIN FETCH sc.tickets "
+                    + "LEFT JOIN FETCH sc.user "
+                    + "WHERE sc.user = :user", ShoppingCart.class)
+                    .setParameter("user", user)
+                    .uniqueResult();
+        } catch (HibernateException e) {
+            throw new DataProcessingExeption("Couldn't get shopping cart id=" + user.getId(), e);
+        }
```

Comment:
Let's remove the catch block and return the Optional ShoppingCart instead

Reply:
We don't need optional here because user has a shopping cart, it is always added to a new user at the moment he registered.
Here is no twisted logic for the user with and without a shopping cart, so we no need it.

Reply:
And how about registering a new cart at the moment of a user registering?
But it's done. What do you mean? Look at the authorization service.